File: src/lib/network/protocol/MessageFactory.dart
import 'Message.dart';
import 'messages/server/LoginFailedMessage.dart';
import 'messages/server/ServerHelloMessage.dart';

class MessageFactory {
  static final messages = {
    20100: () => ServerHelloMessage(),
    20103: () => LoginFailedMessage(),
  };

  static bool hasMessage(int id) {
    return messages.containsKey(id);
  }

  static Message create(int id) {
    return messages[id]();
  }
}

File: src/api/Network/Protocol/MessageFactory.cs
using System;
using System.Collections.Generic;
using SupercellUilityApi.Network.Protocol.Messages.Server;

namespace SupercellUilityApi.Network.Protocol
{
    public class MessageFactory
    {
        public static Dictionary<int, Type> Messages;

        static MessageFactory()
        {
            Messages = new Dictionary<int, Type>
            {
                {20100, typeof(ServerHelloMessage)},
                {20103, typeof(LoginFailedMessage)}
            };
        }
    }
}
